hooks/_common.sh

```diff
@@ -326,8 +360,56 @@ function common::terraform_init {
   fi
 
   if [ ! -d .terraform/modules ] || [ ! -d .terraform/providers ]; then
-    init_output=$(terraform init -backend=false "${TF_INIT_ARGS[@]}" 2>&1)
-    exit_code=$?
+    # Plugin cache dir can't be written concurrently or read during write
+    # https://github.com/hashicorp/terraform/issues/31964
+    if [[ -z $TF_PLUGIN_CACHE_DIR || $parallelism_disabled == true ]]; then
+      init_output=$(terraform init -backend=false "${TF_INIT_ARGS[@]}" 2>&1)
+      exit_code=$?
+    else
+
+      if command -v flock &> /dev/null; then
+        init_output=$(
+          flock --exclusive "$TF_PLUGIN_CACHE_DIR" \
+            terraform init -backend=false "${TF_INIT_ARGS[@]}" 2>&1
+        )
+        exit_code=$?
+      # Fallback to a "simple-lock" mechanism if `flock` is not available
+      else
+
+        # Get lockfile name unique to the current pre-commit process
+        local -r lockfile="/tmp/TF_PLUGIN_CACHE_DIR_lock_$(grep -a -P $PPID | cut -d' ' -f1)"
```

Comment:
I feel like `grep` should be `pgrep` and `-a` should be dropped so that `cut` is not needed:
```suggestion
        local -r lockfile="/tmp/TF_PLUGIN_CACHE_DIR_lock_$(pgrep -P $PPID)"
```